src/main/java/gregtech/common/tileentities/machines/basic/GT_MetaTileEntity_Miner.java

```diff
@@ -67,73 +60,23 @@ public boolean hasFreeSpace() {
+                mineBlock(aBaseMetaTileEntity, drillX, drillY, drillZ);
+                return true;
             }
-            if (mProgresstime == SPEED[mTier] - 1) {
-                if (isPickingPipes) {
-                    if (drillY == 0) {
-                        aBaseMetaTileEntity.disableWorking();
-                        isPickingPipes = false;
-                    } else if (aBaseMetaTileEntity.getBlockOffset(0, drillY, 0) == MINING_PIPE_TIP_BLOCK || aBaseMetaTileEntity.getBlockOffset(0, drillY, 0) == MINING_PIPE_BLOCK) {
-                        mOutputItems[0] = MINING_PIPE.copy();
-                        mOutputItems[0].stackSize = 1;
-                        aBaseMetaTileEntity.getWorld().setBlockToAir(aBaseMetaTileEntity.getXCoord(), aBaseMetaTileEntity.getYCoord() + drillY, aBaseMetaTileEntity.getZCoord());
-                        drillY++;
-                    }
-                    return;
-                }
-                if (drillY == 0) {
-                    moveOneDown(aBaseMetaTileEntity);
-                    return;
-                }
-                if (drillZ > RADIUS[mTier]) {
-                    moveOneDown(aBaseMetaTileEntity);
-                    return;
-                }
-                while (drillZ <= RADIUS[mTier]) {
-                    while (drillX <= RADIUS[mTier]) {
-                        Block block = aBaseMetaTileEntity.getBlockOffset(drillX, drillY, drillZ);
-                        int blockMeta = aBaseMetaTileEntity.getMetaIDOffset(drillX, drillY, drillZ);
-                        if (block instanceof GT_Block_Ores_Abstract) {
-                            TileEntity tTileEntity = getBaseMetaTileEntity().getTileEntityOffset(drillX, drillY, drillZ);
-                            if (tTileEntity != null && tTileEntity instanceof GT_TileEntity_Ores && ((GT_TileEntity_Ores) tTileEntity).mNatural) {
-                                mineBlock(aBaseMetaTileEntity, drillX, drillY, drillZ);
-                                return;
-                            }
-                        } else {
-                            ItemData association = GT_OreDictUnificator.getAssociation(new ItemStack(block, 1, blockMeta));
-                            if (association != null && association.mPrefix.toString().startsWith("ore")) {
-                                mineBlock(aBaseMetaTileEntity, drillX, drillY, drillZ);
-                                return;
-                            }
-                        }
-                        drillX++;
-                    }
-                    drillX = -RADIUS[mTier];
-                    drillZ++;
-                }
+        } else {
+            ItemData association = GT_OreDictUnificator.getAssociation(new ItemStack(block, 1, blockMeta));
+            if (association != null && association.mPrefix.toString().startsWith("ore")) {
```

Comment:
Немного странно, что _"ore"_ за столько лет не вынесли в константу )
```suggestion
            if (doesBlockHaveOreAssociation(block, blockMeta)) {
            ...
            private bool doesBlockHaveOreAssociation(Block block, int blockMeta)
            {
                ItemData association = GT_OreDictUnificator.getAssociation(new ItemStack(block, 1, blockMeta));
                return association != null && association.mPrefix.toString().startsWith("ore");
            }
```

Reply:
а смысл констранты из 3х букв? думаешь это может изменится в будущем ?

Reply:
Это скорее nit, просто выразил удивление, что нет стандарта для наименований.